simulator/modules/decode/bypass/data_bypass.h

```diff
@@ -149,7 +149,7 @@ void DataBypass<ISA>::trace_new_register( const Instr& instr, Register num)
         entry.next_stage_after_first_execution_stage.set_to_writeback();
 
 
-    if ( !instr.is_bypassible())
+    if ( !instr.is_bypassible() || num.is_mips_hi())
```

Comment:
I think it should be localized into `is_bypassible`.

Reply:
I am sorry, do you mean that it should be `if ( !instr.is_bypassible())` ? 
If I put the check into `is_bypassible` method of perf_instr, the LO register will be not ready for bypass in `multiplication/division` instructions as well, because `trace_new_register` method is called for both of destinations. `is_bypassible` can be specialized in order to work out the problem, so it will look like `if ( !instr.is_bypassible( num))`, which is a little bit ugly. I can also introduce a new method like `has_two_valid_dsts` to `perf_instr` in order to avoid using methods which seem to be ISA specific in DataBypass class, but be able to treat instructions in `trace_new_instr` method in accordance with their needs. 

Should I go with the one which affects the LO register or work out the solution which is devoid of any possible checks in DataBypass class for the HI register or mips multiplication(for example, `is_mips_divmult` or `has_two_valid_dsts` in `perf_instr` class)?

Reply:
> Do you mean that it should be if ( !instr.is_bypassible()) ?

I would prefer that.

> trace_new_register method is called for both of destinations

I would call it only for first destination.
There should be an alternative method for second destination.
Is that possible?